### PR TITLE
Fix bugs in lazy relaying

### PR DESCRIPTION
**Pull request by jcorbin**
With tests to demonstrate the stack traces reported in #29.

r @kriskowal @Raynos

### CODE REVIEW

**Comment by jcorbin** [comment]
The only question I really have is do we want to keep the dead warning logs as is for now, or work to silence them?  See travis output or run test locally to see more, but basically it's things like:
```
# relay request times out
TCHANNEL WARN: dropping error from dead relay request ~ { error:
   { [TchannelRequestTimeoutError: request timed out after 96ms (limit was 93ms)]
     type: 'tchannel.request.timeout',
...
ok 17 expected timeout error
ok 18 expected no response
TCHANNEL WARN: dropping error from dead relay request ~ { error:
   { [TchannelLocalResetError: tchannel: tchannel: Connection was manually closed.]
     type: 'tchannel.local.reset',
...
# relay request declines on no peer
TCHANNEL WARN: no relay peer available ~ { requestType: 'tchannel.lazy.incoming-request',
  inRemoteAddr: '127.0.0.1:56144',
  inRequestId: 2,
  serviceName: 'dest' }
ok 19 expected declined error
ok 20 expected no response
ok 21 cluster has an empty state
# relay request handles channel close correctly
TCHANNEL WARN: dropping error from dead relay request ~ { error:
   { [TchannelLocalResetError: tchannel: tchannel: Connection was manually closed.]
     type: 'tchannel.local.reset',
...
```

File: relay_handler.js [hunk 1]
```diff
@@ -221,23 +221,16 @@ function relayRequestlogError(err, codeName) {
     });
 };
 
-LazyRelayInReq.prototype.checkTimeout =
```
Comment (Raynos) [inline relay_handler.js]: where did `checkTimeout` go ?

**Comment by Raynos** [comment]
Looks good enough.

File: relay_handler.js (hunk 1 above)
Comment (jcorbin) [inline relay_handler.js]: it's now onTimeout, checkTimeout was never right for these classes.

**Comment by kriskowal** [comment]
:ok:

**Pull request merged**